libdnf/rpm/nevra.cpp

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright (C) 2018-2020 Red Hat, Inc.
+
+This file is part of libdnf: https://github.com/rpm-software-management/libdnf/
+
+Libdnf is free software: you can redistribute it and/or modify
+it under the terms of the GNU Lesser General Public License as published by
+the Free Software Foundation, either version 2 of the License, or
+(at your option) any later version.
+
+Libdnf is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU Lesser General Public License for more details.
+
+You should have received a copy of the GNU Lesser General Public License
+along with libdnf.  If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#include "libdnf/rpm/nevra.hpp"
+
+#include <regex>
+
+namespace libdnf::rpm {
+
+
+#define PKG_NAME    "([^:(/=<> ]+)"
+#define PKG_EPOCH   "(([\\][*?0-9]+):)?"
+#define PKG_VERSION "([^-:(/=<> ]+)"
+#define PKG_RELEASE PKG_VERSION
+#define PKG_ARCH    "([^-:.(/=<> ]+)"
+
+static const std::regex NEVRA_FORM_REGEX[]{
```

Comment:
I know. We agreed to use the "clang-format" to ensure the consistency of code format.
But the NEVRA_FORM_REGEX[] table has been specially formatted for good readability. Output after using of "clang-format" is ugly.
Please switch off clang-format for the table.
`// clang-format off`
And switch it back on after the table definition.
`// clang-format on`